text/0162-refactoring-support.md

```diff
@@ -0,0 +1,549 @@
+(from `MyStack/Bucket/Resource` to `Web/Website/Origin/Resource` etc.) Since the
+CDK computes the logical IDs of the resources based on their path in the tree,
+all three resources will have different logical IDs in the synthesized template,
+compared to what is already deployed.
+
+If you run `cdk deploy` now, by default the CLI will detect these changes and
+present you with a selection prompt:
+
+    The following resources were moved or renamed:
+
+    ┌───────────────────────────────┬───────────────────────────────┬──────────────────────────┐
+    │ Resource Type                 │ Old Logical ID                │ New Logical ID           │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::S3::Bucket               │ MyStack.Queue1A4198146        │ Web.Bucket843D52FF       │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::CloudFront::Distribution │ MyStack.DistributionE3BB089E  │ Web.Distribution7142E1F1 │
+    ├───────────────────────────────┼───────────────────────────────┼──────────────────────────┤
+    │ AWS::Lambda::Function         │ MyStack.FunctionA5EA2BD8      │ Service.Function8F0BB69B │
+    └───────────────────────────────┴───────────────────────────────┴──────────────────────────┘
+
+    ? What do you want to do? (Use arrow keys)
+    ❯ Execute the refactor and deploy
+      Deploy without refactoring (will cause resource replacement)
+      Quit
+
+If you choose to refactor and deploy, the CLI will show the progress as the
+refactor is executed:
+
+    Refactoring...
+    Creating stack refactor...
+
+     0/3 | 2:03:17 PM | REFACTOR_IN_PROGRESS | AWS::S3::Bucket               | MyStack.Bucket5766466B       | Web.Bucket843D52FF           
+     0/3 | 2:03:17 PM | REFACTOR_IN_PROGRESS | AWS::CloudFront::Distribution | MyStack.DistributionE3BB089E | Web.Distribution7142E1F1     
+     1/3 | 2:03:18 PM | REFACTOR_COMPLETE    | AWS::S3::Bucket               | MyStack.Bucket5766466B       | Web.Bucket843D52FF           
+     1/3 | 2:03:18 PM | REFACTOR_IN_PROGRESS | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.Function8F0BB69B     
+     2/3 | 2:03:19 PM | REFACTOR_COMPLETE    | AWS::CloudFront::Distribution | MyStack.DistributionE3BB089E | Web.DistributionE3BB089E 
+     3/3 | 2:03:20 PM | REFACTOR_COMPLETE    | AWS::Lambda::Function         | MyStack.FunctionA5EA2BD8     | Service.FunctionA5EA2BD8     
+    
+    ✅  Stack refactor complete
+
+You can configure the refactoring behavior by passing the `--refactoring` flag
+to the CLI, or by configuring this setting in the `cdk.json` file:
+
+```json
+{
+  "app": "...",
+  "refactoring": "EXECUTE_AND_DEPLOY"
+}
+```
+
+### Ambiguity
+
+In the unlikely event that there are two or more _identical_ resources (that is,
+resources with the same properties, except for the `Metadata` field), and you
```

Comment:
same properties do you mean same property field and value combinations?
e.g. if i have a stack of lambda function with loggroup pointing to a logical id of a log group resource. and i refactor this entire stack, will cfn refactoring api handle it?

Reply:
Created an appendix to explain more precisely what it means for two resources to be equivalent.